primefaces/src/main/java/org/primefaces/component/avatar/AvatarRenderer.java

```diff
@@ -145,11 +145,16 @@ protected String calculateLabel(FacesContext context, Avatar avatar) {
     /**
      * Generates a dynamic color based on the hash of the label.
      *
-     * @param label the label to generate the color for
+     * @param avatar to generate the color for
      * @return the new color and background color styles
      */
-    protected String generateBackgroundColor(String label) {
-        return "color:#fff;background-color: hsl(" + Math.abs((label.hashCode() % 40) * 9) + ", 100%, 50%);";
+    protected String generateBackgroundColor(Avatar avatar) {
+        boolean withAlpha = avatar.getAlpha() != null;
+        return "background-color:hsl" + (withAlpha ? "a" : Constants.EMPTY_STRING) + "(" +
+                Math.abs((avatar.getLabel().hashCode() % 40) * 9) +
+                "," + avatar.getSaturation() + "%" +
+                "," + avatar.getLightness() + "%" +
+                (withAlpha ? "," + avatar.getAlpha() + "%" : Constants.EMPTY_STRING) + ");";
```

Comment:
Just to clean this code up a little maybe you should use `String.format()` to make it easier to read and pass variables to?